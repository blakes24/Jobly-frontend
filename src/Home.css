.Home {
  text-align: center;
  background-color: burlywood;
  height: calc(100vh - 56px);
  background-image: url("./guy.jpg");
  background-size: cover;
  background-position: center;
}

.Home h1 {
  padding-top: 15vh;
  font-size: 3rem;
  font-weight: bold;
  color: #025fc2;
}

.Home p {
  font-size: 18px;
}
