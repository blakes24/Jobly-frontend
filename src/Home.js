import { useContext } from 'react';
import UserContext from './UserContext';

const Home = () => {
  const user = useContext(UserContext);
  return (
    <div>
      <h1>Jobly</h1>
      <p>Find your next job.</p>
      {user && <p>Welcome back {user.username}!</p>}
    </div>
  );
};

export default Home;
