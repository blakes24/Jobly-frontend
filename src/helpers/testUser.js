import UserContext from './UserContext';

const testUser = {
  username     : 'testuser',
  firstName    : 'Test',
  lastName     : 'User',
  email        : 'joel@joelburton.com',
  isAdmin      : false,
  applications : []
};

const UserProvider = ({ children, user = testUser }) => (
  <UserContext.Provider value={{ user }}>{children}</UserContext.Provider>
);

export { UserProvider };
