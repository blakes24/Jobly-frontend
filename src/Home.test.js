import { render } from '@testing-library/react';
import { UserProvider } from './testUser';
import Home from './Home';

test('renders without crashing', () => {
  render(
    <UserProvider>
      <Home />
    </UserProvider>
  );
});

it('matches snapshot logged out', function() {
  const { asFragment } = render(
    <UserProvider user={null}>
      <Home />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot logged in', function() {
  const { asFragment } = render(
    <UserProvider>
      <Home />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
