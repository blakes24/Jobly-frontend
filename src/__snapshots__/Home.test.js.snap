// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot logged in 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <p>
      The best place to find your next job.
    </p>
    <h2
      class="Home-welcome"
    >
      Welcome back testuser!
    </h2>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot logged out 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <p>
      The best place to find your next job.
    </p>
  </div>
</DocumentFragment>
`;
